app/models/ability.rb

```diff
@@ -264,6 +285,17 @@ def initialize(user)
         user.in?(submission.tutorial.tutors)
       end
 
+      can :fill_tag_select, Tag
+
+      # only generic users who are speakers can assemble the talk
+      can [:assemble, :modify], Talk do |talk|
+        talk.given_by?(user) && talk.visible_for_user?(user)
+      end
+
+      cannot :show, Talk do |talk|
+        !talk.visible_for_user?(user)
+      end
```

Comment:
I think it would be better to whitelist, then to blacklist. This file is in need for refactoring.

Reply:
The whole structure and logic of ability.rb will be rebuilt in an upcoming PR (including this line).